### PR TITLE
Remove cluster from kubeconfig when deleting

### PR DESCRIPTION
**Pull request by kschumy**
### Description
If the kubeconfig contains an eksctl-created cluster, deleting the cluster will now remove the cluster from the kubeconfig.

### Checklist
- [x] Code compiles correctly (i.e `make build`)
- [x] Added tests that cover your change (if possible)
- [x] All tests passing (i.e. `make test`)
- [ ] Added/modified documentation as required (such as the README)
- [x] Added yourself to the `humans.txt` file

### CODE REVIEW

**Comment by richardcase** [comment]
> I'll work on tests and get back to you

Great, thanks @kschumy.  The tests for the kubeconfig package use standard Go tests but for other packages we have started to use [Gingko](https://onsi.github.io/ginkgo/) (for example, the `ami` package).  Its easier to stick with the current test approach (i.e. standard go test) with this package but if you feel like going down the Ginkgo route then thats good as well.

Let me know if i can help.

**Comment by kschumy** [comment]
Sorry! I didn't see your comment until now. I created a test but did not use Ginkgo, but I'll do that in the future

File: pkg/utils/kubeconfig/kubeconfig_test.go [hunk 1]
```diff
@@ -225,3 +274,71 @@ users:
...
+		}
+		os.Remove(tempKubeconfigPath)
+	}, nil
+}
+
+// Cluster names are 'cluster-one.us-west-2.eksctl.io' and 'cluster-two.us-west-2.eksctl.io'.
+// All the information for cluster-one.us-west-2.eksctl.io is identical to one-cluster.yaml. If all
+// the information for cluster-two.us-west-2.eksctl.io is deleted, the file should be identical to
+// one-cluster.yaml and oneClusterAsBytes.
+var twoClustersAsBytes, _ = ioutil.ReadFile("./two-clusters.yaml")
```
Comment (richardcase) [inline pkg/utils/kubeconfig/kubeconfig_test.go]: With the move of the test files:

`ioutil.ReadFile("./testdata/two-clusters.yaml")`

File: pkg/utils/kubeconfig/kubeconfig_test.go [hunk 2]
```diff
@@ -225,3 +274,71 @@ users:
...
+
+// Cluster name is 'cluster-one.us-west-2.eksctl.io'.
+// All the information is identical to cluster cluster-one.us-west-2.eksctl.io in two-clusters.yaml.
+var oneClusterAsBytes, _ = ioutil.ReadFile("./one-cluster.yaml")
```
Comment (richardcase) [inline pkg/utils/kubeconfig/kubeconfig_test.go]: Like above:

`ioutil.ReadFile("./testdata/one-cluster.yaml")`

File: pkg/utils/kubeconfig/kubeconfig_test.go (hunk 1 above)
Comment (kschumy) [reply pkg/utils/kubeconfig/kubeconfig_test.go]: Oh wow 😱 thanks!

**Pull request merged**